### ISSUE TITLE
help to newbies which not similar with npm

### ISSUE THREAD

**Issue by AntonSikidin** [plexsystems/rabbit-viz#20]
Hello, i am trying to install and use this package. I am not familiar with npm.

Can you provide step by step instruction how use it on linux pc without node.js and npm.

I am run "npm install plexsystems/rabbit-viz" and script collect over 250mb other modules
npm run start  do nothing
cd node_modules/rabbit-viz/ 
npm run start  throw errors

> rabbit-viz@1.0.0 start /home/noone/waiter2call/node_modules/rabbit-viz
> react-scripts start

It looks like you're trying to use TypeScript but do not have typescript installed.
Please install typescript by running npm install typescript.
If you are not trying to use TypeScript, please remove the tsconfig.json file from your package root (and any TypeScript files).

npm ERR! Linux 4.15.0-46-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "start"
npm ERR! node v8.10.0
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! rabbit-viz@1.0.0 start: `react-scripts start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the rabbit-viz@1.0.0 start script 'react-scripts start'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the rabbit-viz package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     react-scripts start
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs rabbit-viz
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls rabbit-viz
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/noone/waiter2call/node_modules/rabbit-viz/npm-debug.log
noone@harorah:~/waiter2call/node_modules/rabbit-viz$ 

can you help me how to use it?

**Comment by jpreese** [comment]
Thanks for reaching out, @AntonSikidin. I agree, we should probably update our documentation to include how to get rabbit-viz up and running from a clean state.

To get this project to work locally, you should just need to do the following:

1. git clone https://github.com/plexsystems/rabbit-viz.git
(this will download the source to the current working directory, a folder named **rabbit-viz** should appear)

2. cd rabbit-viz

3. npm install
(this will install all of the required modules and dependencies. it might take a little bit, so just wait until the downloads are complete)

4. npm run start
(this will start up a local copy of the rabbit-viz project. A browser should appear and rabbit-viz should be visible. If for some reason it's not, try navigating to http://localhost:3000 after rabbit-viz has finished the run command)

Alternatively, if you're just looking to use rabbit-viz, this repository hosts a working version at https://plexsystems.github.io/rabbit-viz/

**Comment by AntonSikidin** [comment]
thanks, it works,  
another question: where i can generate json from my rabbitmq server to view?

**Comment by AntonSikidin** [comment]
previously i think that  rabbit-viz get data from running server

**Comment by AntonSikidin** [comment]
answer 
http://localhost:15672/#/ ->overview ->Export definitions->Download broker definitions

**Comment by numo16** [comment]
Here's the docs on the RabbitMQ site for exporting definitions: https://www.rabbitmq.com/backup.html#definitions-export

TL;DR: You can retrieve the definitions for your server a couple of ways:

1. Make a `GET` request against the management api at `/api/definitions`
2. Download it from the management UI overview pane
![export-definitions](https://user-images.githubusercontent.com/268388/55251047-005f6f00-5226-11e9-8fb4-bd66a358cc25.PNG)

**Comment by AntonSikidin** [comment]
feature request:

add button to load definition from local server

**Comment by AntonSikidin** [comment]
http://localhost:15672/api/definitions  work fine, but atomaticaly it runs better
